text/0000-network-spec.md

```diff
@@ -80,6 +136,27 @@ In pseudo code, this would look like:
     sleep(wait_time)
 ```
 
+### API
+
+- **Add epoch_hash/height info:** Add information of all validators that belong to the same shard as we do for every epoch.
+- **Start active epoch_hash/height:** Active epoch are usually current and next epoch. When new epoch gets active this service will start to establish connection between peers in that epoch.
+- **Finish active epoch_hash/height:** This epoch_hash already finished so, connections made because of are marked as not important anymore, so can be dropped if the routing table grows
+
+### Policy
+
+- At all moments we should try to have a minimum number of outbound connections.
+- We will accept (or try) connections to other validators in our same shard.
```

Comment:
shards -> each validator can be allocated to multiple shards.
Also only block + chunk producers really need to be fully interconnected.

"hidden" validators just need a sparse connecitivity.
Also non-validators are "tracking" subset of shards -> shards they want to have state for.
Non-validators don't need to connect to every single block/chunk producer either. But should be connected to at least to one node that tracks shard they are tracking and with reduced distance to chunk producers.

Reply:
That is what I meant for block+chunk producers.

What about this tracking subset of shards? How do you imagine this subset will be selected, I mean do we care really about tracking some shards or is it ok to have sparse connectivity in general?